cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -101,10 +115,51 @@ int MetricsCollector::RecordMetrics(
     AWS_LOGSTREAM_DEBUG(__func__, "Recording metric with name=[" << metric_msg->metric_name << "]");
 
     // create a MetricObject with message parameters to batch
-    Aws::CloudWatchMetrics::Utils::MetricObject metric_object {metric_msg->metric_name,
+    Aws::CloudWatchMetrics::Utils::MetricObject metric_object{metric_msg->metric_name,
                                                               metric_msg->value,
                                                               metric_msg->unit,
-                                                              GetMetricDataEpochMillis(*metric_msg),
+                                                              GetMetricDataEpochMillis(metric_msg->time_stamp),
+                                                              dimensions,
+                                                              this->storage_resolution_.load()};
+    bool batched = metric_service_->batchData(metric_object);
+
+    if (!batched) {
+      AWS_LOGSTREAM_ERROR(__func__, "Failed to record metric");
+    }
+
+    batched_count++;
+  }
+  return batched_count;
+}
+
+int MetricsCollector::RecordMetrics(
+  metrics_statistics_msgs::msg::MetricsMessage::UniquePtr msg)
+{
+  int batched_count = 0;
+  AWS_LOGSTREAM_DEBUG(__func__, "Received " << msg->statistics.size() << " metrics");
+
+  for (auto & statistic : msg->statistics) {
+    std::map<std::string, std::string> dimensions;
+    if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_AVERAGE) {
+      dimensions["statistic_type"] = "avg";
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_MAXIMUM) {
+      dimensions["statistic_type"] = "max";
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_MINIMUM) {
+      dimensions["statistic_type"] = "min";
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_SAMPLE_COUNT) {
+      dimensions["statistic_type"] = "count";
+    } else if (statistic.data_type == StatisticDataType::STATISTICS_DATA_TYPE_STDDEV) {
+      dimensions["statistic_type"] = "stddev";
+    }
+    dimensions["metric_type"] = msg->metrics_source;
+
+    AWS_LOGSTREAM_DEBUG(__func__, "Recording metric with name=[" << msg->measurement_source_name << "]");
+
+    // create a MetricObject with message parameters to batch
+    Aws::CloudWatchMetrics::Utils::MetricObject metric_object{msg->measurement_source_name,
+                                                              statistic.data,
+                                                              "",
```

Comment:
```suggestion
                                                              "" /* unit */,
```

Reply:
I have added a `unit` field to MetricsMessage (https://github.com/ros-tooling/system_metrics_collector/pull/85), so it isn't really necessary to add this comment anymore.